spec/upload.md

```diff
@@ -0,0 +1,318 @@
+# upload@2.0 protocol
+
+## Abstract
+
+This specifiaction defines protocol for _access_ and _store_ services as set of operations in [IPLD Schema][] which can be executed via [UCAN invocation]. This has following implications:
+
+1. Operations can be encoded as [IPLD][] blocks.
+2. Operations can be addressed by a derived, globaly unique, identifiers - [CID][]s.
+3. Requests _(containing multiple operations)_ can be efficiently transported using Content Archives ([CAR][]s) omitting redundant blocks (e.g. serivie issued UCANs, proofs that had already been transported).
+4. Can work over libp2p or other long lived connections.
+
+
+## Transport
+
+Protocol is designed with a single HTTP endpoint in mind which client can `POST` requests with arbitrary number of operations packed as a single [CAR][] file.
+
+
+That said specification is not tied to this design choice. Implementation MAY choose to encode operations as HTTP requests targeting different endpoints. Alternatively protocol may be exposed using GraphQL interface. More broadly as long as requests can be parsed into data structures corresponding to [operations](#Operations) transport format should not matter.
+
+## Requests
+
+Requests are represented as [UCAN invocation][]s in [IPLD][] encoding. Therefor they:
+
+1. MAY contain one or more operations _(represented as [capabilities][UCAN capabilities])_.
+2. MUST have cryptographic proofs of access.
+
+
+Request is defined as a following `UCAN` type in [IPLD schema][]:
+
+```ipldsch
+type UCAN struct {
+  capabilities [Capability]
+  issuer DID
+  audience DID
+  expires optional Time
+  begins optional Time
+  proofs [&UCAN]
+}
+
+
+type Capability union {
+  -- account access service
+  | Identify  "access/identify"
+  | Authorize "access/authorize"
+  | Revoke    "access/revoke"
```

Comment:
Actually UCAN spec has separate representation for revocation so we should use that instead of having our separate thing here
https://github.com/ucan-wg/spec/#57-revocation